receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -466,3 +489,24 @@ func (s *scraper) scrapeAndAppendSignalsPendingMetric(ctx context.Context, now p
 
 	return nil
 }
+
+func (s *scraper) getWMIParentPidFunc() parentPidFunc {
+	return func(_ context.Context, _ processHandle, pid int32) (int32, error) {
+		ppid64, err := s.wmiProcInfoManager.GetProcessPpid(int64(pid))
+		if err != nil {
+			return 0, err
+		}
+		return int32(ppid64), nil
+	}
+}
+
+func configureWindowsSettings(s *scraper) error {
+	if s.config.DisableWMI {
+		if s.config.Metrics.ProcessHandles.Enabled {
+			return errProcessHandlesRequiresWMI
```

Comment:
This should be checked in `Config::Validate` to fail early and have the collector's `validate` command use it.